controllers/proverbs-controllers.js

```diff
@@ -0,0 +1,73 @@
+const { validationResult } = require('express-validator');
+
+const Proverb = require('../models/proverb');
+const user = require('../models/user');
+
+const getProverbs = async (req, res, next) => {
+	res.send('hi from prov bro ;) ');
+};
+
+const postProverb = async (req, res, next) => {
+	const errors = validationResult(req);
+	if (!errors.isEmpty()) {
+		return next(errors);
+	}
+
+	const { proverb, literalTranslation: translation, explanation, userId } = req.body;
+
+	const postedProverb = new Proverb({
+		proverb,
+		translation,
+		explanation,
+		userId
+	});
+
+	if (userId) {
```

Comment:
In this part of the code, you could take advantage of a technique called return early pattern:

https://medium.com/swlh/return-early-pattern-3d18a41bba8#:~:text=Return%20early%20is%20the%20way,when%20conditions%20are%20not%20met.

https://szymonkrajewski.pl/why-should-you-return-early/

```
if (!userId) {
   try {
			await postedProverb.save();
		} catch (error) {
			console.log(error);
			res.status(500).json({
				msg: 'Could not save user in database'
			});
			return next(error);
		}
		
     return res.status(201).json({ proverb: postedProverb });
}

		let user;

		try {
			user = await user.findById(userId);
		} catch (error) {
			console.log(error);
			res.status(500).json({
				msg: 'Could not find user in database'
			});
			return next(error);
		}

		if (!user) {
			res.status(500).json({
				msg: 'Could not find user'
			});
			throw new Error('Invalid credentials.');
		}
  
		try {
			const session = await mongoose.startSession();
			session.startTransaction();
			await postedProverb.save({ session });
			user.proverbs.push(postedProverb);
			await user.save({ session });
			await sess.commitTransaction();
		} catch (error) {
			res.status(500).json({
				msg: 'Could not save proverb for user'
			});
			return next(error);
		}
		
    return res.status(201).json({ user_proverbs: user.proverbs });
```

When we remove levels of indentations from our code, it becomes easier to read for the others.